app/controllers/concerns/client_access_control_concern.rb

```diff
@@ -4,9 +4,13 @@ module ClientAccessControlConcern
   private
 
   def require_client_login
-    unless current_client.present?
+    if current_client.blank?
       session[:after_client_login_path] = request.original_fullpath if request.get?
-      redirect_to new_portal_client_login_path
+      redirect_to new_portal_client_login_path and return
+    end
+
+    if current_client.triggered_still_needs_help_at.present?
```

Comment:
Consider extracting the still_needs_help logic into its own method due to single responsibility principle of methods -- method name is require_client_login, but this is now also doing a redirect to a still needs help flow for clients who are logged in.

Reply:
we just reviewed this - seems like a good idea and will refactor after merging this